celery/backends/database/session.py

```diff
@@ -50,7 +55,26 @@ def create_session(self, dburi, short_lived_sessions=False, **kwargs):
 
     def prepare_models(self, engine):
         if not self.prepared:
-            ResultModelBase.metadata.create_all(engine)
+            # SQLAlchemy will check if the items exist before trying to
+            # create them, which is a race condition. If it raises an error
+            # in one iteration, the next may pass all the existence checks
+            # and the call will succeed.
+            max_retries = 10
+            retries = 0
+            while True:
+                try:
+                    ResultModelBase.metadata.create_all(engine)
```

Comment:
So subsequent calls to `.create_all` will succeed, even if the schema has been already created?

Reply:
Yes, which is currently what happens since each celery worker calls `.create_all` unconditionally. See the checkfirst arg [in the docs](https://docs.sqlalchemy.org/en/13/core/metadata.html#sqlalchemy.schema.MetaData.create_all).